src/renderer.ts

```diff
@@ -153,6 +156,430 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+    const padding = context.option('rowPadding', 1);
+
+    return {
+
+      template: `<g class="heatmapcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const rect = d3.select(n).selectAll('rect').data(col.getValue(d.v, d.dataIndex));
+
+        rect.enter().append('rect');
+        rect
+          .attr({
+            y: padding,
+            x: (d, i) => (i * celldimension),
+            width: celldimension,
+            height: context.rowHeight(i),
+            fill: (d, i) => multiValueColumn.getColor(d)
+          });
+        rect.exit().remove();
+      }
+    };
+  }
+
+
+  createHTML(col: Column, context: IDOMRenderContext): IHTMLCellRenderer {
+    const multiValueColumn: MultiValueColumn = (<MultiValueColumn>col);
+    const celldimension = multiValueColumn.calculateCellDimension();
+    const padding = context.option('rowPadding', 1);
+
+    return {
+      template: `<div class="heatmapcell" style="top:${padding}px;">
+                                   </div>`,
+      update: (n: HTMLDivElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const div = g.selectAll('div').data(col.getValue(d.v, d.dataIndex));
+        div.enter().append('div');
+        div
+          .style({
+            'width': celldimension + 'px',
+            'background-color': multiValueColumn.getColor(d),
+            'height': context.rowHeight(i) + 'px',
+            'left': (d, i) => (i * celldimension) + 'px'
+          });
+
+
+      }
+    };
+  }
+
+  createCanvas(col: Column, context: ICanvasRenderContext): ICanvasCellRenderer {
+    const multiValueColumn: MultiValueColumn = (<MultiValueColumn>col);
+    const celldimension = multiValueColumn.calculateCellDimension();
+    const padding = context.option('rowPadding', 1);
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+
+      var data = col.getValue(d.v, d.dataIndex);
+      data.forEach(function (d, i) {
+
+        var x = (i * celldimension);
+        ctx.beginPath();
+        ctx.fillStyle = multiValueColumn.getColor(d);
+        ctx.fillRect(x, padding, celldimension, context.rowHeight(i));
+      });
+    };
+  }
+
+}
+
+
+class SparklineCellRenderer implements ICellRendererFactory {
+
+
+  createSVG(col: Column, context: IDOMRenderContext): ISVGCellRenderer {
+    const multiValueColumn: MultiValueColumn = (<MultiValueColumn>col);
+    return {
+
+      template: `<g class="sparklinecell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const path = d3.select(n).selectAll('path').data([<any>col.getValue(d.v, d.dataIndex)]);
+        var line = d3.svg.line<number>();
+        path.enter().append('path');
+        path
+          .attr('d', function (d, i) {
+            line
+              .x((d, i) => multiValueColumn.getxScale(i))
+              .y((d, i) => multiValueColumn.getyScale(d, context.rowHeight(i)));
```

Comment:
there are three `i` at this place, defined in lines 242, 248, 250 ... which one should be send to `context.rowHeight`?